src/autopas/containers/cellPairTraversals/C01BasedTraversal.h

```diff
@@ -32,11 +31,14 @@ class C01BasedTraversal : public CBasedTraversal<ParticleCell, PairwiseFunctor,
    * @param pairwiseFunctor The functor that defines the interaction of two particles.
    * @param interactionLength Interaction length (cutoff + skin).
    * @param cellLength cell length.
+   * @param dataLayout The data layout with which this traversal should be initialised.
+   * @param useNewton3 Parameter to specify whether the traversal makes use of newton3 or not.
    */
   explicit C01BasedTraversal(const std::array<unsigned long, 3> &dims, PairwiseFunctor *pairwiseFunctor,
-                             double interactionLength, const std::array<double, 3> &cellLength)
-      : CBasedTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3, collapseDepth>(
-            dims, pairwiseFunctor, interactionLength, cellLength) {}
+                             double interactionLength, const std::array<double, 3> &cellLength,
+                             DataLayoutOption::Value dataLayout, bool useNewton3)
```

Comment:
I only see now that you always pass `DataLayoutOption::Value`. Is there a reason? Can't we simply use `DataLayoutOption`? I thought there should be automatic conversion between the wrapper and value but correct me if I'm wrong.

Reply:
Seems to work. I probably adopted the `::Value` from the templates.

Reply:
That's interesting: Using the Value operator of the inherited member `_dataLayout` in a switch-statement, seems to cause problems with clang, but not with gcc. 
Error message: `error: case value is not a constant expression`
If you add a `getValue()` method to `DataLayoutOption`, or an explicit `this->` it works.

```C++
// in DataLayoutOption.h
constexpr operator Value() const { return _value; }
constexpr Value getValue() const { return _value; } // added only for testing purpose


// in VVLAsBuildTraversal.h for example
switch (_dataLayout) {...} // not working with clang12, but with gcc11
switch (_dataLayout.getValue()) {...} // working 
switch (this->_dataLayout) {...} // working 
```

Not quite sure if its related to this here: https://stackoverflow.com/questions/71533672/bug-in-clang-12-case-value-is-not-a-constant-expression